The same merged dataset used in the EDA section is recreated here by loading the two CSV files, cleaning game names, and merging them based on title.

In [2]:
import pandas as pd
from scipy.stats import ttest_ind

# Load datasets
file1 = pd.read_csv("bestSelling_games.csv")
file2 = pd.read_csv("steam_games_2026.csv")

# Clean game names for matching
file1["game_name"] = file1["game_name"].str.lower().str.strip()
file2["Name"] = file2["Name"].str.lower().str.strip()

# Merge datasets
df = pd.merge(file1, file2, left_on="game_name", right_on="Name", how="inner")

Null Hypothesis:
There is no significant difference in average downloads across different price categories.

Alternative Hypothesis:
There is a significant difference in average downloads across at least one price category.

In [3]:
from scipy.stats import ttest_ind

free = df[df["price"] == 0]["estimated_downloads"]
paid = df[df["price"] > 0]["estimated_downloads"]

t_stat, p_value = ttest_ind(free, paid)

print("p-value:", p_value)

p-value: 7.736606989726841e-08


To examine whether free and paid games differ in popularity, an independent t-test was conducted.

The test produced a p-value of 7.73 × 10⁻⁸.

Since this value is significantly less than 0.05, the null hypothesis is rejected. This indicates that there is a statistically significant difference between free and paid games in terms of downloads.

This result supports the earlier analysis, where free games showed substantially higher download counts.

Null Hypothesis: Genre and downloads are independent

Alternative Hypothesis: Genre affects downloads

In [4]:
df["download_category"] = pd.cut(df["estimated_downloads"],
                                bins=[0, 1e6, 1e7, 1e8],
                                labels=["Low", "Medium", "High"])

In [5]:
from scipy.stats import chi2_contingency

table = pd.crosstab(df["Primary_Genre"], df["download_category"])
chi2, p, _, _ = chi2_contingency(table)

print("p-value:", p)

p-value: 0.1501983131316525


A chi-square test was conducted to examine the relationship between genre and download category. The p-value was 0.1502, which is greater than 0.05. Therefore, we fail to reject the null hypothesis.

This suggests that there is no statistically significant relationship between genre and download category.

However, exploratory data analysis indicates that some genres (such as Action) tend to have higher average downloads. This difference may not be statistically significant due to variability within genres and the categorization of downloads.

Overall, the hypothesis testing results show that price has a statistically significant effect on game downloads, as indicated by the t-test.

However, the chi-square test suggests that genre does not have a statistically significant relationship with download categories.

This contrasts with the EDA findings, where certain genres appeared to have higher average downloads. This difference may be due to variability within genres and the way downloads were categorized.

In conclusion, while price shows a clear statistical impact on game popularity, the effect of genre is less conclusive and may require further analysis.